controllers/configurationpolicy_controller.go

```diff
@@ -1606,21 +1618,42 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 			} else {
 				resultEvent.compliant = true
 				resultEvent.reason = reasonWantNotFoundDNE
+
+				// Compliant, complianceType == mustnothave and the name field is empty
+				// Find resources in the same namespace to add to the status.relatedObjects
+				if objDetails.kind != "" && objDetails.name == "" {
+					if len(allResourceNames) > 0 {
+						shouldAddCondensedRelatedObj = true
+					}
+				}
 			}
 		}
 
 		result = objectTmplEvalResult{objectNames: objNames, events: []objectTmplEvalEvent{resultEvent}}
 
-		relatedObjects = addRelatedObjects(
-			resultEvent.compliant,
-			mapping.Resource,
-			objDetails.kind,
-			namespace,
-			objDetails.isNamespaced,
-			objNames,
-			resultEvent.reason,
-			nil,
-		)
+		if shouldAddCondensedRelatedObj {
+			// Length of objNames = 0, complianceType == musthave and the name field is empty
```

Comment:
This comment is out of date since `mustnothave` can also set `shouldAddCondensedRelatedObj` to `true`.